Propagate input tensor names in XNNPACK backend.

AshkanAliabadi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35896 Add XNNPACK Clamp.
* #35791 Enable stateless XNNPACK linear.
* #35790 Enable stateless XNNPACK convolutions.
* #35354 Support for XNNPACK max pooling operator.
* **#35351 Propagate input tensor names in XNNPACK backend.**

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D20821861](https://our.internmc.facebook.com/intern/diff/D20821861)

Commit:
Propagate input tensor names in XNNPACK backend.

[ghstack-poisoned]

kimishpatel reviewed (commented):
Can you add a comment as to what is this change for and why is it needed?

AshkanAliabadi commented:
@gchanan Hi Greg.  I modified this function to propagate tensor names since I was getting a test failure in test_namedtensor.py for the max pooling changes stacked on top of this PR.  Having said that, after making this change, I now get the following warning (not an error, but still) in tests:

Warning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable.

Does this change look good to you?  Should I do anything differently?

AshkanAliabadi commented on aten/src/ATen/native/xnnpack/Factory.cpp:
```diff
@@ -31,7 +33,9 @@ Tensor empty_with_tail_padding(
-  return tensor.resize_(size, memory_format);
+  return namedinference::propagate_names_if_nonempty(
+      tensor.resize_(size, memory_format),
+      maybe_names);
```
This is the change. cc @gchanan

Commit:
Update on "Propagate input tensor names in XNNPACK backend."




[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."




[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."




[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."




[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."




[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."




[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D20821861](https://our.internmc.facebook.com/intern/diff/D20821861)

[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D20821861](https://our.internmc.facebook.com/intern/diff/D20821861)

[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D20821861](https://our.internmc.facebook.com/intern/diff/D20821861)

[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D20821861](https://our.internmc.facebook.com/intern/diff/D20821861)

[ghstack-poisoned]

Commit:
Update on "Propagate input tensor names in XNNPACK backend."


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D20821861](https://our.internmc.facebook.com/intern/diff/D20821861)

[ghstack-poisoned]
